### PR TITLE
Remove lyft refs imports

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
_Please replace this text with a description of what this PR accomplishes._

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,63 +1,70 @@
...
-	sigs.k8s.io/structured-merge-diff v0.0.0-20190302045857-e85c7b244fd2 // indirect
 )
 
 replace (
 	github.com/GoogleCloudPlatform/spark-on-k8s-operator => github.com/lyft/spark-on-k8s-operator v0.1.4-0.20201027003055-c76b67e3b6d0
-	github.com/lyft/flytestdlib => github.com/flyteorg/flytestdlib v0.3.13
-	github.com/lyft/flyteidl => github.com/flyteorg/flyteidl v0.18.15
+	github.com/aws/amazon-sagemaker-operator-for-k8s => github.com/aws/amazon-sagemaker-operator-for-k8s v1.0.1-0.20210303003444-0fb33b1fd49d
+
+	github.com/flyteorg/flyteidl => github.com/flyteorg/flyteidl v0.18.15
```
Comment (kumare3) [inline go.mod:61]: we dont need this anymore

File: go.mod [hunk 2]
```diff
@@ -1,63 +1,70 @@
...
+	github.com/flyteorg/flytestdlib => github.com/flyteorg/flytestdlib v0.3.13
```
Comment (kumare3) [inline go.mod:62]: same

File: go.mod [hunk 3]
```diff
@@ -1,63 +1,70 @@
...
+
+	github.com/golang/protobuf => github.com/golang/protobuf v1.3.5
```
Comment (kumare3) [inline go.mod:64]: why?

File: go.mod [hunk 4]
```diff
@@ -1,63 +1,70 @@
...
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.3.1
+	google.golang.org/genproto => google.golang.org/genproto v0.0.0-20200205142000-a86caf926a67
```
Comment (kumare3) [inline go.mod:66]: why?

File: go.mod [hunk 5]
```diff
@@ -1,63 +1,70 @@
...
+	google.golang.org/grpc => google.golang.org/grpc v1.28.0
+	k8s.io/client-go => k8s.io/client-go v0.20.2
```
Comment (kumare3) [inline go.mod:68]: why not directt

File: go.mod [hunk 6]
```diff
@@ -1,63 +1,70 @@
...
-	github.com/lyft/flyteidl => github.com/flyteorg/flyteidl v0.18.15
+	github.com/aws/amazon-sagemaker-operator-for-k8s => github.com/aws/amazon-sagemaker-operator-for-k8s v1.0.1-0.20210303003444-0fb33b1fd49d
+
+	github.com/flyteorg/flyteidl => github.com/flyteorg/flyteidl v0.18.15
+	github.com/flyteorg/flytestdlib => github.com/flyteorg/flytestdlib v0.3.13
+
+	github.com/golang/protobuf => github.com/golang/protobuf v1.3.5
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.3.1
+	google.golang.org/genproto => google.golang.org/genproto v0.0.0-20200205142000-a86caf926a67
+	google.golang.org/grpc => google.golang.org/grpc v1.28.0
```
Comment (kumare3) [inline go.mod:67]: why not direct

File: go.mod (hunk 3 above)
Comment (katrogan) [reply go.mod:64]: for reasons I don't understand, I can't actually set the protobuf version to 1.3.5 above -> go mod always updates it to v1.4.3 despite my best efforts 🤷

File: go.mod (hunk 5 above)
Comment (katrogan) [reply go.mod:64]: I can't! it keeps getting replaced

File: go.mod (hunk 4 above)
Comment (katrogan) [reply go.mod]:
```
github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery/core imports
	github.com/flyteorg/flytestdlib/storage imports
	github.com/graymeta/stow/google imports
	cloud.google.com/go/storage imports
	cloud.google.com/go tested by
	cloud.google.com/go.test imports
	google.golang.org/genproto/googleapis/cloud/secretmanager/v1: module google.golang.org/genproto@latest found (v0.0.0-00010101000000-000000000000, replaced by google.golang.org/genproto@v0.0.0-20200205142000-a86caf926a67), but does not contain package google.golang.org/genproto/googleapis/cloud/secretmanager/v1
```

**Comment by kumare3** [comment]
cc @regadas this is the update, also i will FYI you on the Propeller PR

File: go/tasks/plugins/k8s/sidecar/sidecar.go [hunk 7]
```diff
@@ -65,14 +65,39 @@ func validateAndFinalizePod(
 	return &pod, nil
 }
 
-func (sidecarResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCore.TaskExecutionContext) (k8s.Resource, error) {
-	sidecarJob := plugins.SidecarJob{}
+// Why, you might wonder do we recreate the generated go struct generated from the plugins.SidecarJob proto? Because
+// although we unmarshal the task custom json, the PodSpec itself is not generated from a  proto definition,
+// but a proper go struct defined in k8s libraries. Therefore we only unmarshal the sidecar as a json, rather than jsonpb.
+type sidecarJob struct {
+	PodSpec              *k8sv1.PodSpec
```
Comment (kumare3) [inline go/tasks/plugins/k8s/sidecar/sidecar.go:72]: do you want to add `json:..` tags? - so that it remains

**Review by kumare3 — APPROVED** [review]

**Pull request merged**